### DISCUSSION TITLE
Using configuration package locally

### DISCUSSION THREAD

**Discussion by Sir-gits-a-lot** [crossplane/crossplane#4638] [Q&A]
How can I install a packaged configuration in a k8s cluster directly using the .xpkg file? Actually I have restriction to upload it to OCI registry but I wanna test locally. I am running a kind cluster and have installed crossplane on it. Also, Do I need to also package the provider config along with provider ?

**Comment by turkenh** [comment]
This could be tricky unless you push it to a registry. 

There are two different approaches we are following today to achieve this:

1. [Mount a PVC](https://github.com/crossplane-contrib/provider-aws/blob/29651ece73cdd17c6ce9a76ae054c273e1018ef7/cluster/local/integration_tests.sh#L93-L168) to Crossplane with the xpkg in the cache and configure XP accordingly.
2. [Patch a Sidecar](https://github.com/upbound/build/blob/a6e25afa0d43da62b11af96a5d29627a52f32cd9/makelib/local.xpkg.mk#L20) to Crossplane pod and copy with xpkg into the cache directory.

We are successfully using the latter in all Upbound [official providers](https://github.com/upbound/provider-aws/blob/b815340dade4c8a2bef8edf9bf6ed21fb45a315a/Makefile#L199) and most [reference platforms](https://github.com/upbound/platform-ref-aws/blob/c972e0cd2fe91ffdb0b181250832fae74e473b10/Makefile#L70) to test/validate local changes later with uptest.

**Reply by Sir-gits-a-lot** [reply]
I wanna try the first approach in crossplane deployed on a kind cluster. Can you guide me how I can mount the .xpkg file to the node of the kind cluster and do I have to mount it to worker node or control plane node.

**Reply by turkenh** [reply]
Please check the same script, it has all necessary steps including the Kind config with [extraMounts](https://github.com/crossplane-contrib/provider-aws/blob/29651ece73cdd17c6ce9a76ae054c273e1018ef7/cluster/local/integration_tests.sh#L71-L84).

**Reply by Sir-gits-a-lot** [reply]
I'm sorry, which script are you referring to ?
Can you please share the link to the script, if you have it?

**Reply by turkenh** [reply]
The one already linked to `extraMounts` keyword above, https://github.com/crossplane-contrib/provider-aws/blob/29651ece73cdd17c6ce9a76ae054c273e1018ef7/cluster/local/integration_tests.sh#L75

**Reply by Sir-gits-a-lot** [reply]
Can you tell the exact directly in worker node we have to mount pvc to in kind crossplane cluster to make it available as a precached image?

I am docker cp command to copy the .xpkg file to the worker node of kind crossplane cluster but it seems to not work. 

The integration script.sh is too overwhelming for me I am not able follow.
